src/print.rs

```diff
@@ -0,0 +1,185 @@
+/// Prints to the standard output.
+///
+/// Equivalent to the [`println!`] macro except that a newline is not printed at
+/// the end of the message.
+///
+/// Note that stdout is frequently line-buffered by default so it may be
+/// necessary to use [`io::stdout().flush()`][flush] to ensure the output is emitted
+/// immediately.
+///
+/// Use `print!` only for the primary output of your program. Use
+/// [`eprint!`] instead to print error and progress messages.
+///
+/// [`println!`]: macro.println.html
+/// [flush]: io/trait.Write.html#tymethod.flush
+/// [`eprint!`]: macro.eprint.html
+///
+/// # Panics
+///
+/// Panics if writing to `io::stdout()` fails.
+///
+/// # Examples
+///
+/// ```ignore
+/// # async_std::task::block_on(async {
+/// #
+/// use async_std::prelude::*;
+/// use async_std::io;
+/// use async_std::print;
+///
+/// print!("this ").await;
+/// print!("will ").await;
+/// print!("be ").await;
+/// print!("on ").await;
+/// print!("the ").await;
+/// print!("same ").await;
+/// print!("line ").await;
+///
+/// io::stdout().flush().await.unwrap();
+///
+/// print!("this string has a newline, why not choose println! instead?\n").await;
+///
+/// io::stdout().flush().await.unwrap();
+/// #
+/// # })
+/// ```
+#[macro_export]
+macro_rules! print {
+    ($($arg:tt)*) => (
+        async {
+            use ::async_std::prelude::*;
```

Comment:
I'm not sure if this PR is good. This will make it difficult to reexport these macros.

Reply:
Maybe my understanding of scoping is off here; but doesn't this include the prelude only within the `async` block?

The goal of this crate is to provide re-exports for async-std's submodules, because we can't define the macros inside the crate itself. I think as long as we cover that use case we should be okay?

Reply:
I think this is enough for that purpose. 

(I was thinking about the restriction that `$crate` cannot be used. But given proc-macro always has this limitation, we probably don't need to worry.)